pom.xml

```diff
@@ -204,7 +204,8 @@
           <target>1.8</target>
           <compilerArgs>
             <compilerArg>-Werror</compilerArg>
-            <compilerArg>-Xlint:deprecation</compilerArg>
+            <!-- protobuf generates deprecated code for top level enums -->
+            <!--<compilerArg>-Xlint:deprecation</compilerArg>-->
```

Comment:
Isn't it possible to exclude this code somehow without turning off deprecation checking for the whole code base? We can't be the first who came across this issue.

Reply:
I did not find any recipe for maven, see
https://github.com/google/protobuf/issues/2054

Reply:
There's a couple of approaches possible; None is pretty, but it's prettier than turning off all warnings. I would actually go for c), because at some point we should break the bookkeeper client out of the server package, and in this case, the protocol will be a shared dependency.

a) use maven-replacer to add SuppressWarnings to all the generated classes
b) use a different compile phase for generated code.
c) put all generated code in a separate module

https://stackoverflow.com/questions/39994647/maven-3-how-to-exclude-generated-sources-from-xlint-check
https://github.com/google/protobuf/issues/1596

Reply:
@ivankelly I prefer c) too, I have created the issue. I will change this after the issue is checked in.
https://github.com/apache/bookkeeper/issues/691

The weird think is that for @sijie the warning does not appear. Maybe he did not run the protoc compiler on this version. The bug is only for top-level enums, not for inner classes

Reply:
+1 for 3)